joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -524,6 +540,26 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
     return CallbackReturn::FAILURE;
   }
 
+  if (has_velocity_command_interface_ && use_closed_loop_pid_adapter)
+  {
+    pids_.resize(joint_names_.size());
+    velocity_ff_.resize(joint_names_.size());
+
+    // Init PID gains from ROS parameter server
+    for (size_t i = 0; i < pids_.size(); ++i)
+    {
+      const std::string prefix = "gains." + joint_names_[i];
+      const auto k_p = auto_declare<double>(prefix + ".p", 0.0);
+      const auto k_i = auto_declare<double>(prefix + ".i", 0.0);
+      const auto k_d = auto_declare<double>(prefix + ".d", 0.0);
+      const auto i_clamp = auto_declare<double>(prefix + ".i_clamp", 0.0);
+      velocity_ff_[i] = auto_declare<double>("velocity_ff/" + joint_names_[i], 0.0);
+      // Initialize PID
+      pids_[i] = std::make_shared<control_toolbox::Pid>(k_p, k_i, k_d, i_clamp, -i_clamp);
+    }
+    // TODO(livanov93): add other option when implemented
```

Comment:
what other option(s)?

Reply:
@destogl  Was alluding on effort interface, [here](https://github.com/livanov93/ros2_controllers/blob/2d2970d01cdaee11387b92bee090a668c084f489/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L543)  we will have to leave only `use_closed_loop_pid_adapter` variable in the if condition. 
Wrong choice of word - change when implemented would be more appropriate.

Reply:
ros-controls/ros2_controllers@68d838d